### PR TITLE
Test/xiaolei/td 13708 test csharp subscribe

### PR DESCRIPTION
**Pull request by xleili**

### CODE REVIEW

File: src/connector/C#/src/test/FunctionTest/SubscribeAsync.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                        for (int i = 0; i < actualResData.Count; i++)
+                        {
+                            // output.WriteLine("index:{0},expectResData:{1},actualResData:{2}", i, expectResData[i], actualResData[i]);
+                            Assert.Equal(expectResData2[i].ToString(), actualResData[i]);
+                        }
+                    }
+                }
+                else
+                {
+                    output.WriteLine($"async query data failed, failed code {code}");
```
Comment (sangshuduo) [inline src/connector/C#/src/test/FunctionTest/SubscribeAsync.cs]: how about "error code" here? and can we provide more info provided by taos_errstr() interface?

**Review by sangshuduo — COMMENTED** [review]
see my comments

File: src/connector/C#/src/test/FunctionTest/SubscribeAsync.cs (hunk 1 above)
Comment (xleili) [reply src/connector/C#/src/test/FunctionTest/SubscribeAsync.cs]: fixed

File: src/connector/C#/src/test/FunctionTest/SubscribeAsync.cs [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                            // output.WriteLine("index:{0},expectResData:{1},actualResData:{2}", i, expectResData[i], actualResData[i]);
+                            Assert.Equal(expectResData[i].ToString(), actualResData[i]);
+                        }
+                    }
+
+
+                }
+                else if (taosRes != IntPtr.Zero)
+                {
+                    output.WriteLine($"async query data failed, failed code {TDengine.Error(taosRes)}");
```
Comment (sangshuduo) [inline src/connector/C#/src/test/FunctionTest/SubscribeAsync.cs]: should be: error code xx, reason: xxxxx

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

**Review by zitsen — APPROVED** [review]
LGTM

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**